cpp/src/parquet/stream_reader.cc

```diff
@@ -0,0 +1,224 @@
+#include "parquet/stream_reader.h"
+
+namespace parquet {
+
+StreamReader::StreamReader(std::unique_ptr<ParquetFileReader>& File_Reader)
+    : file_reader_{File_Reader.release()} {
+  file_metadata_ = file_reader_->metadata();
+
+  auto* const schema{file_metadata_->schema()};
+  auto* const group_node{schema->group_node()};
+
+  nodes_.resize(schema->num_columns());
+
+  for (auto i = 0; i < schema->num_columns(); ++i) {
```

Comment:
int64_t?

Reply:
The SchemaDescriptor::num_columns() function returns an integer so I do not think that is necessary.

Reply:
Please try to avoid `auto` where possible. We strive to have the type of the variable at least once in the line it gets defined (for readability reasons).